prefect_aws/deployments/steps.py

```diff
@@ -117,7 +117,9 @@ def push_to_s3(
             continue
         elif not local_file_path.is_dir():
             remote_file_path = Path(folder) / local_file_path.relative_to(local_path)
-            client.upload_file(str(local_file_path), bucket, str(remote_file_path))
+            client.upload_file(
+                str(local_file_path), bucket, str(remote_file_path.as_posix())
```

Comment:
This trick can help ensure it works on Windows and Linux platforms.
```suggestion
                str(local_file_path), bucket, str(PureWindowsPath(remote_file_path.as_posix()))
```

Reply:
@desertaxle I think it's a problem in the test, not in the code.

I ran the following code on both windows and linux:

```
from pathlib import Path
folder = 'arbitrary/s3/folder/'
local_path = Path.cwd()
for local_file_path in local_path.expanduser().rglob("*"):
    if not local_file_path.is_dir():
        remote_file_path = Path(folder) / local_file_path.relative_to(local_path)
        print(f"not_posix: {str(remote_file_path)}")
        print(f"yes_posix: {str(remote_file_path.as_posix())}")
```

Windows yielded:

```
not_posix: arbitrary\s3\folder\top.txt
yes_posix: arbitrary/s3/folder/top.txt
not_posix: arbitrary\s3\folder\dir1\sub.txt
yes_posix: arbitrary/s3/folder/dir1/sub.txt
```

Linux yielded:

```
not_posix: arbitrary/s3/folder/top.txt
yes_posix: arbitrary/s3/folder/top.txt
not_posix: arbitrary/s3/folder/dir1/sub.txt
yes_posix: arbitrary/s3/folder/dir1/sub.txt
```

So I think that the code that I wrote will not break based on OS, but I might need to make the mock of the returned files more accurate. Thoughts? Am I missing something?

Reply:
@desertaxle done and done